src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/SmeltersTool.java

```diff
@@ -0,0 +1,86 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.tools;
+
+import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
+import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.attributes.DamageableItem;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ToolUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
+import io.github.thebusybiscuit.slimefun4.utils.tags.SlimefunTag;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import org.bukkit.Effect;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Collection;
+import java.util.Optional;
+
+public class SmeltersTool extends SimpleSlimefunItem<ToolUseHandler> implements DamageableItem {
+
+    @ParametersAreNonnullByDefault
+    public SmeltersTool(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(itemGroup, item, recipeType, recipe);
+    }
+
+    @Override
+    public @Nonnull
+    ToolUseHandler getItemHandler() {
+        return (e, tool, fortune, drops) -> {
+            Block b = e.getBlock();
+
+            if (isValid(tool, b)) {
+                Collection<ItemStack> blockDrops = b.getDrops(tool);
+
+                for (ItemStack drop : blockDrops) {
+                    if (drop != null && !drop.getType().isAir()) {
+                        smelt(b, drop, fortune);
+                        drops.add(drop);
+                    }
+                }
+
+                damageItem(e.getPlayer(), tool);
+            }
+        };
+    }
+
+    private boolean isValid(@Nonnull ItemStack tool, @Nonnull Block b) {
+        if (BlockStorage.hasBlockInfo(b)) {
+            return false;
+        } else {
+            Material toolMaterial = tool.getType();
+            Material blockMaterial = b.getType();
+            if (SlimefunTag.PICKAXES.isTagged(toolMaterial) && SlimefunTag.SMELTERS_PICKAXE_BLOCKS.isTagged(blockMaterial)) {
+                return true;
+            } else if (SlimefunTag.SHOVELS.isTagged(toolMaterial) && SlimefunTag.SMELTERS_SHOVEL_BLOCKS.isTagged(blockMaterial)) {
+                return true;
+            } else if (SlimefunTag.AXES.isTagged(toolMaterial) && SlimefunTag.SMELTERS_AXE_BLOCKS.isTagged(blockMaterial)) {
```

Comment:
you dont want a shovel to work on iron ore do you?
it would be more weird if it did imo

Reply:
I do agree the ifs are super weird. Merging the statements becomes a bit of an unreadable mess. But As jeff says, you kinda need to ensure a Pickaxe doesn't smelt a log or sand etc - otherwise the purpose of having the other tools is kinda moot (other than the speed). Could be done with a tag for all smeltables with a minable pickaxe tag check, but that feels worse I think?

Reply:
yea you got Block#isPreferredTool()

Reply:
Maybe switch statements are possible instead seeing as SlimefunTag is an enumeration.

Reply:
> yea you got Block#isPreferredTool()

so this is a simple
if (isPrefferedTool() && smeltable) {
}

Reply:
im confused, what shoukd i do?

Reply:
Add a new tag being the combination of the 3 smeltable tags you’ve already made (having them available as separate could be useful for other things) called SMELTABLE_BLOCKS or something. Then replace isValid() with a b.isPrefferedTool(tool) && SlimefunTag.SMELTABLE_BLOCKS.isTagged(b.getType) ? Something like that anyway. You could just make it one tag from the get-go also.

Reply:
*cries in polymorphism*

Reply:
done